packages/math-poolsimulator/src/gyro2/index.test.ts

```diff
@@ -0,0 +1,63 @@
+import { bnum, SubgraphPoolBase } from "@balancer-labs/sor";
+import { describe, expect, test } from "@jest/globals";
+
+import { numberToBigNumber } from "../conversions";
+import { AMM } from "../index";
+import poolsFromFile from "./fixtures/data.json";
+import { ExtendedGyro2 } from "./index";
+
+describe("Tests new Gyro 2CLP math function based on package other functions", () => {
+  const poolData = poolsFromFile["pool"];
+
+  const tokenIn = "USDC";
+  const tokenOut = "DAI";
+
+  const amm = new AMM(
+    new ExtendedGyro2({
+      swapFee: poolData.swapFee,
+      totalShares: poolData.totalShares,
+      tokens: poolData.tokens,
+      tokensList: poolData.tokensList,
+      sqrtAlpha: poolData.sqrtAlpha,
+      sqrtBeta: poolData.sqrtBeta,
+    }),
+  );
+
+  const pool = ExtendedGyro2.fromPool(poolData as SubgraphPoolBase);
+  const poolPairData = pool.parsePoolPairData(tokenIn, tokenOut);
+  const amount = 50742;
+  const amountOldBigNumber = bnum(amount);
+  const spotPriceExpected = pool._spotPriceAfterSwapExactTokenInForTokenOut(
+    poolPairData,
```

Comment:
Just curious, did we ever tested what happens if we pass 0 to this function?

Reply:
<img width="725" alt="image" src="https://github.com/bleu-studio/balancer-tools/assets/63298341/3446a21b-69b9-415d-8e73-40d1bcbb2e2b">
 😭 it works just like our spot price kkkkkk

Reply:
will open a new PR to solve this